reservation_units/utils/reservation_unit_reservation_state_helper.py

```diff
@@ -90,119 +120,132 @@ def __active_pricing_type_alias(cls) -> dict[str, Subquery]:
-                or active_price.pricing_type == PricingType.FREE
-            )
-        )
+        if active_price is None:
+            return False
+
+        # Pricing is free
+        if active_price.pricing_type == PricingType.FREE:
+            return True
+        # Pricing is paid and there is a payment product
+        else:
+            return reservation_unit.payment_product is not None
 
     @classmethod
     def __has_valid_pricing_query(cls) -> Q:
-        return Q(active_pricing_type=PricingType.PAID, payment_product__isnull=False) | Q(
-            active_pricing_type=PricingType.FREE
+        return (
+            # Pricing is paid and there is a payment product
+            Q(active_pricing_type=PricingType.PAID, payment_product__isnull=False)
+            # or pricing is free
+            | Q(active_pricing_type=PricingType.FREE)
         )
 
     @classmethod
     def __is_reservable(cls, reservation_unit: ReservationUnit) -> bool:
-        """
-        Support following cases in regard to reservation_begins and reservation_ends:
-
-        -------------------------------
-                  |--------------------
-                        |--------------
-        ------|   |--------------------
-        ------|         |--------------
-                        ↑ now
-        """
-        now = datetime.datetime.now(tz=get_default_timezone())
+        now = local_datetime()
 
         return cls.__is_reservable_period(now, reservation_unit) and cls.__has_valid_pricing(reservation_unit)
 
-    @classmethod
-    def __is_reservable_period(cls, now: datetime.datetime, reservation_unit: ReservationUnit) -> bool:
-        return (
-            # Reservations don't begin or end
-            (reservation_unit.reservation_ends is None and reservation_unit.reservation_begins is None)
-            # Reservations don't end, but begin somewhere in the past (or right now)
-            or (
-                reservation_unit.reservation_ends is None
-                and reservation_unit.reservation_begins is not None
-                and reservation_unit.reservation_begins <= now
-            )
-            # Reservations begin somewhere in the past (or right now),
-            # and end in the past, but begin after they end
-            or (
-                reservation_unit.reservation_ends is not None
-                and reservation_unit.reservation_begins is not None
-                and reservation_unit.reservation_ends < reservation_unit.reservation_begins <= now
-            )
-        )
-
     @classmethod
     def __get_is_reservable_query(cls) -> QueryState:
         return QueryState(
             aliases=cls.__active_pricing_type_alias(),
             filters=(cls.__is_reservable_period_query() & cls.__has_valid_pricing_query()),
         )
 
+    @classmethod
+    def __is_reservable_period(cls, now: datetime.datetime, reservation_unit: ReservationUnit) -> bool:
+        """
```

Comment:
Could add a written explanation like the ones above.

Reply:
Will add better documentation to all these when rewriting the helpers to be clearer a later date :)